app/index.js

```diff
@@ -70,7 +72,18 @@ app.on('ready', () => {
 
     // config changes
     const cfgUnsubscribe = config.subscribe(() => {
+      const cfg_ = plugins.getDecoratedConfig();
+
       win.webContents.send('config change');
+
+      if (cfg_.shell !== cfg.shell) {
+        notify(
+          'Shell configuration changed!',
+          'Open a new tab or window to start using the new shell'
+        );
+      }
+
+      cfg = cfg_;
```

Comment:
I'm tempted to leave this as is, as it will update the config for *any* new session. Right now, config updates only affect the react app, but not the node/electron side. This will update the config that gets used for any new session/window.

Also, this is updating the shell for *new* sessions only, not current ones. Any existing sessions will continue to use the one originally loaded when the session was created.